x/model/keeper/msg_server_model.go

```diff
@@ -149,7 +149,7 @@ func (k msgServer) UpdateModel(goCtx context.Context, msg *types.MsgUpdateModel)
 	}
 
 	model.EnhancedSetupFlowOptions = msg.EnhancedSetupFlowOptions
```

Comment:
Also the following check needs to be added according to the spec:
Update Model: if EnhancedSetupFlowTCRevision is present, it must be equal to oldValue+1. If the oldValue == 0, then new value can be any. If the newValue  ==0, then no updates (OK). If the newValue == oldValue - error (must be oldValue + 1)

Reply:
Added validation